apm-agent-plugin-sdk/src/main/java/co/elastic/apm/agent/sdk/weakmap/WeakMapSupplier.java

```diff
@@ -36,12 +36,13 @@
     private static final WeakConcurrentSet<WeakConcurrentSet<?>> registeredSets = new WeakConcurrentSet<>(WeakConcurrentSet.Cleaner.INLINE);
 
     public static <K, V> WeakConcurrentMap<K, V> createMap() {
-        WeakConcurrentMap<K, V> result = new WeakConcurrentMap<>(false);
+        WeakConcurrentMap<K, V> result = new NullSafeWeakConcurrentMap<>(false);
         registeredMaps.add(result);
         return result;
     }
 
     public static <V> WeakConcurrentSet<V> createSet() {
+        // note: we don't have a NullSafeWeakConcurrentSet here because we can't change the underlying map impl.
```

Comment:
Why do we need to change the underlying map and not extend and override set API - `add`, `contains`, `remove` etc.?

Reply:
The constructor does not allow to change the single field, thus that would mean re-implementing the whole class, while it's only about ~120 lines I'm not sure it's really worth it for now.